### PR TITLE
Aws Parameter Store Integration and keycloak version change

### PR DESCRIPTION
**Pull request by sanketshah1997**
## Description

<Please include a brief description of the code changes>

## Issue

Fixes #<issue_number>

## Checklist
- [x] I have commented my code in hard-to-understand areas
- [x] I have created relevant Unit Tests for the feature I implemented
- [x] I have pulled and merged the latest code and resolved all merge conflicts
- [x] I have added someone to review this PR, and informed the team on Slack

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,18 +12,30 @@
 	<artifactId>jjcsa</artifactId>
 	<version>0.0.1-SNAPSHOT</version>
 	<name>jjcsa</name>
-	<description>Demo project for Spring Boot</description>
+	<description>JJC SA Website using Spring Boot</description>
 
 	<properties>
 		<java.version>1.8</java.version>
-		<keycloak.version>11.0.2</keycloak.version>
+		<keycloak.version>13.0.1</keycloak.version>
```
Comment (shah-tejas) [inline pom.xml:19]: To be in sync, could you also update keycloak version in [docker-compose.yml](https://github.com/JJCSA/Authentication/blob/92b6a715d14b3a14e445dcbaeefad4db5b10485b/docker-compose.yml#L17) and also have Dushyant use the same keycloak version in infra?

File: pom.xml [hunk 2]
```diff
@@ -12,18 +12,30 @@
...
 		<org.projectlombok.version>1.18.12</org.projectlombok.version>
 		<springdoc-openapi.version>1.5.5</springdoc-openapi.version>
 		<spring-cloud-aws-version>2.3.1</spring-cloud-aws-version>
+		<spring-cloud-starter-version>3.0.3</spring-cloud-starter-version>
+		<spring-cloud-aws-parameter-store-config-version>2.2.6.RELEASE</spring-cloud-aws-parameter-store-config-version>
 	</properties>
 
 	<dependencies>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
```
Comment (shah-tejas) [inline pom.xml:30]: So the spring cloud version that is compatible with Spring Boot 2.4.x is now maintained under a different groupId: https://awspring.io/learn/introduction/#versions

We should use the dependency coming from  `<groupId>io.awspring.cloud</groupId>` as this is the latest one which will be supported with future versions of spring boot. Also read [this](https://spring.io/blog/2021/03/17/spring-cloud-aws-2-3-is-now-available) blog for more info.

It should be done similar to `spring-cloud-aws-autoconfigure`:
https://github.com/JJCSA/backend/blob/d668f33a9577d589aef7720c0afaa566f4b67b74/pom.xml#L80

**Review by shah-tejas — CHANGES_REQUESTED** [review]
Use the paramstore dependency from: https://mvnrepository.com/artifact/io.awspring.cloud/spring-cloud-aws-parameter-store-config/2.3.1

Derive the version value from `${spring-cloud-aws-version}` which is already set to `2.3.1`.

Also, `spring-cloud-starter` can be removed if not used.

**Comment by shah-tejas** [comment]
It might also be better to define the aws cloud dependencies from the pom:

```
<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>io.awspring.cloud</groupId>
			<artifactId>spring-cloud-aws-dependencies</artifactId>
			<version>2.3.0</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
	</dependencies>
</dependencyManagement>
```

File: pom.xml (hunk 1 above)
Comment (Nishit-Gopani) [reply pom.xml:19]: I think Dushyant have updated the version in docker. We are using updated version on infra.

File: pom.xml (hunk 2 above)
Comment (sanketshah1997) [reply pom.xml:30]: Thanks for providing reference to the latest docs. I was referring to old docs. so this starter dependency was needed when we use org.springframework.cloud for aws parameter store.

**Pull request closed without merging**